satellite/satellitedb/dbx/satellitedb.dbx

```diff
@@ -564,3 +564,47 @@ read one (
     where reset_password_token.owner_id = ?
 )
 delete reset_password_token ( where reset_password_token.secret = ? )
+
+
+//--- offer table ---//
+
+model offer (
+	key id
+	
+	field id	serial 
+	field name text ( updatable )
+	field description text ( updatable )
+	field type int ( updatable )
+	
+	field credit int ( updatable ) 
+	field award_credit_duration_days int ( updatable )
+	field invitee_credit_duration_days int ( updatable )
+
+	field redeemable_cap int ( updatable )
+	field num_redeemed int ( updatable, autoinsert )
```

Comment:
I'm assuming here that redeemable_cap is meant to be a limit on how high num_redeemed is allowed to go. Is that accurate? It would be nice if we could add a CHECK constraint on the db table, if so. I don't think you can do that directly through dbx, but as far as I know we don't require that anything actually uses the DDL generated by dbx.

Reply:
I will add the check in the update query in the db implementation